slay/truss_adapter/deploy.py

```diff
@@ -0,0 +1,325 @@
+            .read_text()
+            .splitlines()
+            if not req.strip().startswith("#")
+        )
+    pip_requirements.extend(image.pip_requirements)
+    # `pip_requirements` will add server requirements which give version conflicts.
+    # config.requirements = pip_requirements
+    pip_requirements_file_path = truss_dir / _REQUIREMENTS_FILENAME
+    pip_requirements_file_path.write_text("\n".join(pip_requirements))
+    # TODO: apparently absolute paths don't work with remote build (but work in local).
+    config.requirements_file = _REQUIREMENTS_FILENAME  # str(pip_requirements_file_path)
+    config.system_packages = image.apt_requirements
+    # Assets.
+    assets = slay_config.get_asset_spec()
+    config.secrets = assets.secrets
+    if definitions.BASTEN_APY_SECRET_NAME not in config.secrets:
+        config.secrets[definitions.BASTEN_APY_SECRET_NAME] = "***"
+    else:
+        logging.info(
+            f"Workflows automatically add {definitions.BASTEN_APY_SECRET_NAME} "
+            "to secrets - no need to manually add it."
+        )
+    config.model_cache.models = assets.cached
+    # Metadata.
+    slay_metadata: definitions.TrussMetadata = definitions.TrussMetadata(
+        user_config=slay_config.user_config,
+        stub_cls_to_url=stub_cls_to_url,
+    )
+    config.model_metadata[definitions.TRUSS_CONFIG_SLAY_KEY] = slay_metadata.dict()
+    return config
+
+
+def make_truss(
+    processor_dir: pathlib.Path,
+    workflow_root: pathlib.Path,
+    slay_config: definitions.Config,
+    model_name: str,
+    stub_cls_to_url: Mapping[str, str],
+    maybe_stub_file: Optional[pathlib.Path],
+) -> pathlib.Path:
+    truss_dir = processor_dir / _TRUSS_DIR
+    truss_dir.mkdir(exist_ok=True)
+    config = _make_truss_config(truss_dir, slay_config, stub_cls_to_url, model_name)
+
+    config.write_to_yaml_file(
+        truss_dir / serving_image_builder.CONFIG_FILE, verbose=False
+    )
+
+    # Copy other sources.
+    model_dir = truss_dir / truss_config.DEFAULT_MODEL_MODULE_DIR
+    model_dir.mkdir(parents=True, exist_ok=True)
+    shutil.copy(
+        processor_dir / f"{definitions.PROCESSOR_MODULE}.py",
+        model_dir / _MODEL_CLASS_FILENAME,
+    )
+
+    pkg_dir = truss_dir / truss_config.DEFAULT_BUNDLED_PACKAGES_DIR
+    pkg_dir.mkdir(parents=True, exist_ok=True)
+    if maybe_stub_file is not None:
+        shutil.copy(maybe_stub_file, pkg_dir)
+    # TODO This assume all imports are absolute w.r.t workflow root (or site-packages).
+    # Also: apparently packages need an `__init__`, or crash.
+    _copy_python_source_files(workflow_root, pkg_dir / pkg_dir)
+
+    # TODO Truss package contains this from `{ include = "slay", from = "." }`
+    # pyproject.toml. But for quick dev loop just copy from local.
+    shutil.copytree(
+        os.path.dirname(slay.__file__),
+        pkg_dir / "slay",
+        dirs_exist_ok=True,
+    )
+    return truss_dir
+
+
+def get_api_key_from_truss_config() -> str:
```

Comment:
nit: `get_api_key_from_trussrc` 
truss_config usually means config.yaml and it doesn't include secret values.

Reply:
Haha, I got mislead y `load_config`, but yeah, renaming this to `trussrc` is more precise.